blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/EnumElasticIO.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2013 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.ElasticClientManager;
+import com.rackspacecloud.blueflood.service.ElasticIOConfig;
+import com.rackspacecloud.blueflood.service.RemoteElasticSearchServer;
+import com.rackspacecloud.blueflood.types.EnumMetric;
+import com.rackspacecloud.blueflood.types.IMetric;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.utils.GlobPattern;
+import com.rackspacecloud.blueflood.utils.Metrics;
+import org.elasticsearch.action.bulk.BulkRequestBuilder;
+import org.elasticsearch.action.index.IndexRequestBuilder;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.SearchHit;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.elasticsearch.index.query.QueryBuilders.*;
+
```

Comment:
This is the new class for creating and querying enum indexes.  Note that this solution depends on adding another "metric_metadata_read" alias to point to the new "enums" index (this alias is currently pointing to both the "metric_metadata" and "metric_metadata_v2" indexes), so that when we query the "metric_metadata_read" alias, it will find all metrics in all 3 indexes (metric_metadata, metric_metadata_v2, enums) with the metric names matching the query value.

The dedupResults() method will handle "include_enum_values" by giving results with enum_values fields precedent over those without, when adding to the results hashmap  (with unique key by metric_name) .

Note to @GeorgeJahad, you are actually right here in that we *don't* have to use a new alias for the enums metrics.  Another way to do it is to do two separate search, one against the read alias (metric_metadata_read), and once against the new "enums" index, then combined/union the results into a new hashmap with the results from "enums" taking precedent.  If you would prefer we do it this way instead, please let me know.